tests/unit/composables/Dialog.test.js

```diff
@@ -0,0 +1,75 @@
+import { mount } from '@vue/test-utils';
+import { useDialog } from 'src/composables/Dialog';
+import DialogEvent from 'src/composables/DialogEvent';
+import { vi } from 'vitest';
+
+vi.mock('src/composables/DialogEvent');
+
+describe('Test: Dialog', () => {
+  let wrapper;
+  let subscribe;
+  let unsubscribe;
+
+  /**
+   * Mount component with composable.
+   * @param {Function} callback - Callback use by the component.
+   */
+  function mountComponent(callback = undefined) {
+    subscribe = vi.fn();
+    unsubscribe = vi.fn();
+
+    DialogEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
+    wrapper = mount({
+      components: { useDialog },
+      setup() {
+        return useDialog('test', callback);
+      },
+      template: '<div id="dialog" v-if="show"></div>',
+    });
+    wrapper.vm.show = false;
+  }
+
+  describe('Test function onDialogEvent', () => {
+    it('should not open dialog when the key is invalid', () => {
+      mountComponent();
+
+      DialogEvent.next({ key: 'invalid' });
+
+      expect(wrapper.vm.show).toBeFalsy();
+    });
+
+    it('should open dialog when the key is valid', () => {
+      mountComponent(() => {
+        expect(wrapper.vm.show).toBeTruthy();
+      });
+
+      DialogEvent.next({ key: 'test' });
+    });
+
+    it('should open dialog when the key is valid and event type is close', () => {
```

Comment:
should close dialog when the key is valid and event type is close

Reply:
done